### PR TITLE
Sketch Plugin Compatability

### PR DESCRIPTION
**Pull request by tommoor**
These changes allow the SDK to run in the environment of a Sketch plugin.

### CODE REVIEW

File: src/AbstractCLI/index.js [hunk 1]
```diff
@@ -93,21 +90,28 @@ export default class AbstractCLI implements AbstractInterface {
...
       });
 
+      const stream = JSONStream.parse();
+      stream.on("data", data => {
+        logStdoutData(data);
+        resolve(data);
+      });
+
       abstractCli.stdout
-        .pipe(JSONStream.parse())
```
Comment (tommoor) [inline src/AbstractCLI/index.js]: Pipe is not available in Sketch so we need to forward data events instead

File: src/AbstractCLI/index.js [hunk 2]
```diff
@@ -80,7 +77,7 @@ export default class AbstractCLI implements AbstractInterface {
         : [];
 
       const spawnArgs = [
-        `./${path.relative(this.cwd, this.cliPath)}`,
+        this.cliPath,
```
Comment (tommoor) [inline src/AbstractCLI/index.js]: Relative path resolution is broken in Sketch

File: src/AbstractCLI/index.js [hunk 3]
```diff
@@ -64,10 +64,7 @@ export default class AbstractCLI implements AbstractInterface {
     this.apiUrl = apiUrl;
 
     try {
-      this.cliPath = path.relative(
-        cwd,
-        path.resolve(cwd, locatePath.sync(cliPath))
-      );
+      this.cliPath = path.resolve(cwd, locatePath.sync(cliPath));
```
Comment (tommoor) [inline src/AbstractCLI/index.js]: This appeared to be doubling up on the logic below on L80, so also needed to go

File: src/AbstractCLI/index.js (hunk 2 above)
Comment (amccloud) [reply src/AbstractCLI/index.js]: I originally did this to make snapshots easier but now paths are replaced with a placeholder

File: src/AbstractCLI/index.js [hunk 4]
```diff
@@ -93,21 +90,28 @@ export default class AbstractCLI implements AbstractInterface {
       logSpawn(spawnArgs);
       const abstractCli = spawn(...spawnArgs);
 
-      let stderrBuffer = new Buffer.from("");
```
Comment (amccloud) [inline src/AbstractCLI/index.js]: good catch! We should probably rename `Abstract.Client` to `Abstract.client` to follow the same factory convention. I've also done `new Abstract.Client()` 🙈

**Pull request merged**